File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/norm/UpperAscii.java
package com.freight.norm;

/** Normalizer: upper ascii. */
public final class UpperAscii implements Normalizer {

    @Override
    public String name() {
        return "upper_ascii";
    }

    @Override
    public String apply(String text) {
        StringBuilder out = new StringBuilder(text.length());
        for (int i = 0; i < text.length(); i++) {
            char c = text.charAt(i);
            out.append((c >= 'a' && c <= 'z') ? (char) (c - 32) : c);
        }
        return out.toString();
    }
}

File: freight-triage-polyglot-ledger-recovery/environment/reconcile/internal/normx/upper_ascii.go
package normx

// UpperAscii normalizes with the upper ascii rule.
func UpperAscii(text string) string {
	out := []byte(text)
	for i, c := range out {
		if c >= 'a' && c <= 'z' {
			out[i] = c - 32
		}
	}
	return string(out)
}
